ecommerce_integrations/shopify/invoice.py

```diff
@@ -63,13 +63,23 @@ def set_cost_center(items, cost_center):
 		item.cost_center = cost_center
 
 
-def make_payament_entry_against_sales_invoice(doc, setting, posting_date=None):
+def make_payament_entry_against_sales_invoice(doc, setting, shopify_order, posting_date=None):
 	from erpnext.accounts.doctype.payment_entry.payment_entry import get_payment_entry
+	_bank_account = setting.get('cash_bank_account')
 
-	payment_entry = get_payment_entry(doc.doctype, doc.name, bank_account=setting.cash_bank_account)
+	if len(shopify_order.get('payment_gateway_names')):
+		doc_gatewaty = shopify_order.get('payment_gateway_names')[0]
+
+	if len(setting.get("payment_gateways")):
+		for gateway in setting.get("payment_gateways"):
+			if doc_gatewaty == gateway.get("gateway_name"):
+				_bank_account = gateway.get("gateway_account")
+
+	payment_entry = get_payment_entry(doc.doctype, doc.name, bank_account=_bank_account)
 	payment_entry.flags.ignore_mandatory = True
 	payment_entry.reference_no = doc.name
 	payment_entry.posting_date = posting_date or nowdate()
 	payment_entry.reference_date = posting_date or nowdate()
 	payment_entry.insert(ignore_permissions=True)
 	payment_entry.submit()
+	frappe.db.commit()
```

Comment:
This shouldn't be added 

https://frappeframework.com/docs/v14/user/en/api/database#database-transaction-model